tests/testthat/test_percentile.R

```diff
@@ -0,0 +1,122 @@
+library(testthat)
+library(recipes)
+library(tibble)
+library(modeldata)
+
+data(biomass)
+
+biomass_tr <- biomass[biomass$dataset == "Training",]
+biomass_te <- biomass[biomass$dataset == "Testing",]
+
+test_that('simple percentile trans', {
+  rec <- recipe(~., data = biomass_tr) %>%
+    step_percentile(carbon, sulfur)
+
+  rec_trained <- prep(rec)
+  biomass_tr_baked <- bake(rec_trained, new_data = biomass_tr)
+  biomass_te_baked <- bake(rec_trained, new_data = biomass_te)
+
+  carbon_quantiles <- quantile(
+    biomass_tr$carbon,
+    probs = (0:100)/100,
+    names = TRUE
+  )
+  sulfur_quantiles <- quantile(
+    biomass_tr$sulfur,
+    probs = (0:100)/100,
+    names = TRUE
+  )
+  sulfur_quantiles <- sulfur_quantiles[!duplicated(sulfur_quantiles)]
+
+  expect_equal(
+    approx(carbon_quantiles, y = 0:100, xout = biomass_tr$carbon)$y/100,
+    biomass_tr_baked$carbon
+  )
+  expect_equal(
+    approx(carbon_quantiles, y = 0:100, xout = biomass_te$carbon)$y/100,
+    biomass_te_baked$carbon
+  )
+  sulfur_values <- as.numeric(gsub("%$", "", names(sulfur_quantiles)))
+  expect_equal(
+    approx(sulfur_quantiles, y = sulfur_values, xout = biomass_tr$sulfur)$y/100,
+    biomass_tr_baked$sulfur
```

Comment:
Can you add a test case where there are fewer unique values in the column being transformed than percentiles requested? Maybe make a new variable that is something like `round(biomass$carbon, -1)`.

Reply:
Added in https://github.com/tidymodels/recipes/pull/882/commits/e204adbe4bbcc688d7c78f6001473d6f57647842